src/derivative_wrappers.jl

```diff
@@ -82,6 +82,17 @@ function vecjacobian!(dλ, λ, p, t, S; dgrad=nothing, dy=nothing)
       jacobian!(J, uf, y, f_cache, sensealg, jac_config)
     end
     mul!(dλ',λ',J)
+
+    if dgrad !== nothing
+      @unpack pJ, pf, paramjac_config = S
+      if DiffEqBase.has_paramjac(f)
+        f.paramjac(pJ,y,sol.prob.p,t) # Calculate the parameter Jacobian into pJ
+      else
+        jacobian!(pJ, pf, sol.prob.p, f_cache, sensealg, paramjac_config)
+      end
+      mul!(dgrad',λ',pJ)
```

Comment:
would make more sense to just do multiple vector Jacobian products instead of forming the Jacobian? With autodiff it's equivalent (if we want to not redo the primal), but with numerical it can save memory.

Reply:
Ah right, doing a numerical `v'J`, like getting the gradient of `v'f(u)`. Yeah, that definitely helps with memory.